systemtest/src/main/java/hivemall/systemtest/runner/SystemTestRunner.java

```diff
@@ -0,0 +1,335 @@
+import hivemall.systemtest.model.CreateTableHQ;
+import hivemall.systemtest.model.DropTableHQ;
+import hivemall.systemtest.model.HQ;
+import hivemall.systemtest.model.HQBase;
+import hivemall.systemtest.model.InsertHQ;
+import hivemall.systemtest.model.RawHQ;
+import hivemall.systemtest.model.TableHQ;
+import hivemall.systemtest.model.TableListHQ;
+import hivemall.systemtest.model.UploadFileAsNewTableHQ;
+import hivemall.systemtest.model.UploadFileHQ;
+import hivemall.systemtest.model.UploadFileToExistingHQ;
+import hivemall.systemtest.utils.IO;
+import hivemall.utils.lang.Preconditions;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+public abstract class SystemTestRunner extends ExternalResource {
+    static final Logger logger = LoggerFactory.getLogger(SystemTestRunner.class);
+    @Nonnull
+    final List<HQBase> classInitHqs;
+    @Nonnull
+    final Set<String> immutableTables;
+    @Nonnull
+    final String dbName;
+    @Nonnull
+    final Properties props;
+
+    SystemTestRunner(@CheckForNull SystemTestCommonInfo ci, @CheckForNull String propertiesFile) {
+        Preconditions.checkNotNull(ci);
+        Preconditions.checkNotNull(propertiesFile);
+
+        classInitHqs = new ArrayList<HQBase>();
+        immutableTables = new HashSet<String>();
+        dbName = ci.dbName;
+
+        final String path = "hivemall/" + propertiesFile;
+        try {
+            InputStream is = null;
+            try {
+                props = new Properties();
+                is = new FileInputStream(Resources.getResource(path).getPath());
+                props.load(is);
+            } finally {
+                if (is != null) {
+                    is.close();
+                }
+            }
+        } catch (Exception ex) {
+            throw new IllegalArgumentException("Failed to load properties from " + path + ". "
+                    + ex.getMessage());
+        }
+    }
+
+    @Override
+    protected void before() throws Exception {
+        initRunner();
+        prepareDB(); // initialize database
+    }
+
+    @Override
+    protected void after() {
+        try {
+            resetDB(); // clean up database
+        } catch (Exception ex) {
+            throw new QueryExecutionException("Failed to clean up temporary database. "
+                    + ex.getMessage());
+        } finally {
+            finRunner();
+        }
+    }
+
+    abstract void initRunner();
+
+    abstract void finRunner();
+
+    public void initBy(@Nonnull final HQBase hq) {
```

Comment:
@Lewuathe Several methods of them can be called from the out of package, so just as it is.